pepper-apis/dsm-core/src/test/java/org/broadinstitute/dsm/service/admin/UserAdminTestUtil.java

```diff
@@ -0,0 +1,404 @@
+     */
+    public void addUserRoles(int userId, List<String> roles) {
+        assertInitialized();
+        for (String role : roles) {
+            int roleId = getRoleId(role);
+            UserAdminService.addUserRole(userId, roleId, studyGroupId);
+            addRoleForUser(roleId, userId);
+        }
+    }
+
+    protected void addRoleForUser(int roleId, int userId) {
+        List<Integer> roleIds = createdUserRoles.get(userId);
+        if (roleIds == null) {
+            throw new DsmInternalError("Invalid user: " + userId);
+        }
+        List<Integer> newRoleIds = new ArrayList<>(roleIds);
+        newRoleIds.add(roleId);
+        createdUserRoles.put(userId, newRoleIds);
+    }
+
+    protected void addRoleForUser(String role, int userId) {
+        addRoleForUser(getRoleId(role), userId);
+    }
+
+    /**
+     * Remove role for user
+     * @param role existing user role
+     */
+    public void removeUserRole(int userId, String role) {
+        assertInitialized();
+        int roleId = getRoleId(role);
+        UserAdminService.deleteUserRole(userId, roleId, studyGroupId);
+        removeRoleForUser(roleId, userId);
+    }
+
+    protected void removeRoleForUser(int roleId, int userId) {
+        List<Integer> roleIds = createdUserRoles.get(userId);
+        if (roleIds == null) {
+            throw new DsmInternalError("Invalid user: " + userId);
+        }
+        List<Integer> newRoleIds = new ArrayList<>(roleIds);
+        newRoleIds.removeAll(List.of(roleId));
+        createdUserRoles.put(userId, newRoleIds);
+    }
+
+    protected void removeRoleForUser(String role, int userId) {
+        removeRoleForUser(getRoleId(role), userId);
+    }
+
+    public void verifyUserHasRole(int userId, String role) {
+        assertInitialized();
+        int userRoleId = UserAdminService.getUserRole(userId, getRoleId(role), studyGroupId);
+        Assert.assertNotEquals(-1, userRoleId);
+    }
+
+    public int getRoleId(String roleName) {
+        assertInitialized();
+        Integer roleId = allRoles.get(roleName);
+        if (roleId == null) {
+            throw new DsmInternalError("Invalid role name: " + roleName);
+        }
+        return roleId;
+    }
+
+    protected static Map<String, Integer> getAllRoles() {
+        return inTransaction(conn -> {
+            Map<String, Integer> roles = new HashMap<>();
+            try (PreparedStatement stmt = conn.prepareStatement("SELECT ar.role_id, ar.name FROM access_role ar")) {
+                try (ResultSet rs = stmt.executeQuery()) {
+                    while (rs.next()) {
+                        roles.put(rs.getString(2), rs.getInt(1));
+                    }
+                }
+            } catch (SQLException e) {
+                throw new DsmInternalError("Error getting roles", e);
+            }
+            return roles;
+        });
+    }
+
+    private static int createTestInstance(String instanceName, int studyGroupId) {
```

Comment:
This code could be used to create an instance regardless of test context, so should it just be called `createInstance()`?

Reply:
Will do. I'm typically not too fussy about private method names, but if it helps readability I'm all for it.